pkg/machine/ignition.go

```diff
@@ -203,6 +215,24 @@ func getFiles(usrName string) []File {
 			Mode: intToPtr(420),
 		},
 	})
+
+	// Issue #11489: make sure that we can inject a custom registries.conf
+	// file on the system level.
+	files = append(files, File{
+		Node: Node{
+			Group: getNodeGrp("root"),
+			Path:  "/etc/containers/registries.conf.d/999-podman-machine.conf",
+			User:  getNodeUsr("root"),
+		},
+		FileEmbedded1: FileEmbedded1{
+			Append: nil,
+			Contents: Resource{
+				Source: strToPtr("data:,unqualified-search-registries%3D%5B%22docker.io%22%5D"),
```

Comment:
Non-blocking, this is pre-existing: a helper for `ignitionFileEmbedded(path string, mode int, contents []byte) FileEmbedded1`, or even a `ignitionFile(more params) File` could be much more readable.

Reply:
I concur.  It took me a while to figure out the nature of the encoding.